src/source/PeerConnection/PeerConnection.c

```diff
@@ -39,20 +39,64 @@ STATUS allocateSrtp(PKvsPeerConnection pKvsPeerConnection)
     return retStatus;
 }
 
+typedef struct {
+    UINT32 currentDataChannelId;
+    PKvsPeerConnection pKvsPeerConnection;
+    PHashTable unkeyedDataChannels;
+} AllocateSctpSortDataChannelsData, *PAllocateSctpSortDataChannelsData;
+
+STATUS allocateSctpSortDataChannelsDataCallback(UINT64 customData, PHashEntry pHashEntry)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    PAllocateSctpSortDataChannelsData data = (PAllocateSctpSortDataChannelsData) customData;
+    PKvsDataChannel pKvsDataChannel = (PKvsDataChannel) pHashEntry->value;
+
+    CHK(customData != 0, STATUS_NULL_ARG);
+
+    pKvsDataChannel->channelId = data->currentDataChannelId;
+    CHK_STATUS(hashTablePut(data->pKvsPeerConnection->pDataChannels, pKvsDataChannel->channelId, (UINT64) pKvsDataChannel));
+
+    data->currentDataChannelId += 2;
+
+CleanUp:
+    return retStatus;
+}
+
 STATUS allocateSctp(PKvsPeerConnection pKvsPeerConnection)
```

Comment:
Is there a better way to do this? I have a HashMap that has values like

```
0 => {id: 5}
1 => {id: 0}
2 => {id: 7}
3 => {id: 1}
```

Is there a way I can sort it in place so it would look like? I tried doing a delete/insert while iterating but that breaks things, didn't dig that much deeper.
```
0 => {id: 0}
1 => {id: 1}
5 => {id: 5}
7 => {id: 7}
```

Reply:
Seems you have a small set of items for which you really don't need a hash table. Having a simple array with a max will work just fine. Same with sorting - an entity inserting into an array can simply do sorted insert with pushing other elements up

Reply:
This can get very large, and I would like to have O(1) lookup.

Before the PeerConnection is established we just don't know the IDs yet. I can keep an array just during startup, but it just feels wasteful/more room for error.